scripts/ci-entrypoint.sh

```diff
@@ -223,14 +230,27 @@ install_cloud_provider_azure() {
 
     export CCM_LOG_VERBOSITY="${CCM_LOG_VERBOSITY:-4}"
     echo "Installing cloud-provider-azure components via helm"
+    rm -f "${CLOUD_PROVIDER_HELM_VALUES}"
+    cat <<EOF > "${CLOUD_PROVIDER_HELM_VALUES}"
+infra:
+  clusterName: "${CLUSTER_NAME}"
+cloudControllerManager:
+  replicas: "${CCM_COUNT}"
+  enableDynamicReloading: "${ENABLE_DYNAMIC_RELOADING}"
+  cloudConfig: "${CLOUD_CONFIG}"
+  cloudConfigSecretName: "${CONFIG_SECRET_NAME}"
+  logVerbosity: "${CCM_LOG_VERBOSITY}"
+  clusterCIDR: "${CCM_CLUSTER_CIDR}"
+EOF
+    echo "Using the following cloud-provider-azure helm configuration:"
+    cat "${CLOUD_PROVIDER_HELM_VALUES}"
+    if [ -f "${CLOUD_PROVIDER_HELM_VALUES_CUSTOM_BUILD}" ]; then
+        cat "${CLOUD_PROVIDER_HELM_VALUES_CUSTOM_BUILD}"
```

Comment:
instead of having to cat and rm a file on every run, would it be cleaner to persist this values file in the capz repo and simply use envsubst to sub in the env vars?

Reply:
Yes.

I tried to scope this change down so I could fix the missing `--kubeconfig` issue as quickly as possible w/ local repro data. (I was blocked doing so on my macos).

We could do one of two things:

1) We could remove this intermittent change entirely and trust that the exported "arrays" implementation seems to work via prow, and then implement your more durable solution as a follow-up.
2) Keep these intermittent changes temporarily and implement your more durable solution as a follow-up.

Reply:
My vote is for 1) so we can merge this quickly and backport it with high confidence. I can confirm I can use the "arrays" implementation on Linux and I know it works in prow too, but we should definitely follow up and fix it for macos users.

Reply:
Done, filed https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3292